### PR TITLE
fix typo for interp_v2,test=develop

### PR DESCRIPTION
**Pull request by tink2123**
### PR types
Bug fixes

### PR changes
APIs

### Describe
* support tuple type for scale_factors
* fix typo
* add notes for v2_op
* Align with torch's calculation
* rename UpSample to Upsample

### CODE REVIEW

**Review by LDOUBLEV — DISMISSED** [review]
LGTM

File: python/paddle/nn/functional/common.py [hunk 1]
```diff
@@ -114,6 +115,8 @@ def interpolate(x,
     smoother than corresponding surfaces obtained by bilinear interpolation or
     nearest-neighbor interpolation.
 
+    Area interpolation is same as adaptive avg pooling.
```
Comment (willthefrog) [inline python/paddle/nn/functional/common.py]: elaborate?

Comment (tink2123) [reply python/paddle/nn/functional/common.py]: Area 插值 找不到比较官方的概念介绍

Comment (willthefrog) [reply python/paddle/nn/functional/common.py]: this should be either removed or revised,  as interpolation is definitely not the same as pooling.

Comment (tink2123) [reply python/paddle/nn/functional/common.py]: Ok，has been removed.

**Review by willthefrog — DISMISSED** [review]

File: python/paddle/nn/functional/common.py (hunk 1 above)
Comment (jzhang533) [reply python/paddle/nn/functional/common.py]: please also delete line 91
and implementation codes in line 287, and line 316 ~ line 322
please run a grep on the file

**Comment by jzhang533** [comment]
in_d, in_h, in_w -> depth, height, width  
or
be consistent with NCW, NWC, ...

File: python/paddle/nn/functional/common.py [hunk 2]
```diff
@@ -207,11 +208,11 @@ def interpolate(x,
              when input is a 4-D Tensor and is (out_d, out_h, out_w) when input is a 5-D Tensor. 
              Default: None. If a list, each element can be an integer or a Tensor Variable of shape: [1].
              If a Tensor Variable, its dimensions size should be a 1.
-        scale_factor (float|Tensor|list|None): The multiplier for the input height or width. At
-             least one of :attr:`out_shape` or :attr:`scale_factor` must be set.
-             And :attr:`out_shape` has a higher priority than :attr:`scale_factor`.Has to match input size if it is a list.
+        scale_factor (float|Tensor|list|tuple|None): The multiplier for the input height or width. At
+             least one of :attr:`size` or :attr:`scale_factor` must be set.
+             And :attr:`size` has a higher priority than :attr:`scale_factor`.Has to match input size if it is a list.
```
Comment (jzhang533) [inline python/paddle/nn/functional/common.py]: :attr:`scale_factor`.Has to match input size if it is a list.

是tensor的时候也要match吧。

File: python/paddle/nn/layer/common.py [hunk 3]
```diff
@@ -273,9 +273,9 @@ class UpSample(layers.Layer):
              when input is a 4-D Tensor and is (out_d, out_h, out_w) when input is a 5-D Tensor. 
              Default: None. If a list, each element can be an integer or a Tensor Variable of shape: [1].
              If a Tensor Variable, its dimensions size should be a 1.
-        scale_factor (float|Tensor|list|None): The multiplier for the input height or width. At
-             least one of :attr:`out_shape` or :attr:`scale_factor` must be set.
-             And :attr:`out_shape` has a higher priority than :attr:`scale_factor`.Has to match input size if it is a list.
+        scale_factor (float|Tensor|list|tuple|None): The multiplier for the input height or width. At
+             least one of :attr:`size` or :attr:`scale_factor` must be set.
+             And :attr:`size` has a higher priority than :attr:`scale_factor`.Has to match input size if it is a list.
```
Comment (jzhang533) [inline python/paddle/nn/layer/common.py]: see above

File: python/paddle/nn/layer/common.py [hunk 4]
```diff
@@ -466,9 +466,9 @@ class UpsamplingBilinear2d(layers.Layer):
              layer, the shape is (out_h, out_w) when input is a 4-D Tensor. 
              Default: None. If a list, each element can be an integer or a Tensor Variable of shape: [1].
              If a Tensor Variable, its dimensions size should be a 1.
-        scale_factor (float|int|list|Tensor|None): The multiplier for the input height or width. At
-             least one of :attr:`out_shape` or :attr:`scale_factor` must be set.
-             And :attr:`out_shape` has a higher priority than :attr:`scale_factor`.
+        scale_factor (float|int|list|tuple|Tensor|None): The multiplier for the input height or width. At
+             least one of :attr:`size` or :attr:`scale_factor` must be set.
+             And :attr:`size` has a higher priority than :attr:`scale_factor`.
              Default: None. Has to match input size if it is a list.
```
Comment (jzhang533) [inline python/paddle/nn/layer/common.py]: see above

File: python/paddle/nn/functional/common.py (hunk 1 above)
Comment (tink2123) [reply python/paddle/nn/functional/common.py]: polish the notes, please review~

File: python/paddle/nn/functional/common.py (hunk 2 above)
Comment (tink2123) [reply python/paddle/nn/functional/common.py]: Done, thx

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**